systemtests/src/main/java/io/enmasse/systemtest/scale/ScalePerformanceTestManager.java

```diff
@@ -161,10 +197,56 @@ public void monitorMetrics() {
         }
     }
 
-    private void waitUntilHasValue(Supplier<Optional<Counter>> counterSupplier, String timeoutMessage) {
+    public Duration measureClientsDowntime() {
+        List<Double> averages = new ArrayList<>();
+        for (var client : clientsMap.values()) {
+
+            TestUtils.waitUntilCondition(() -> {
+                var optional = client.getMetricsClient().getReconnectDurationHistogram();
+                return optional.isPresent() && optional.get().getSampleCount() > 0;
+            }, Duration.ofSeconds(25), Duration.ofSeconds(5), () -> {
+                logger.info("Client {} not reporting reconnections maybe it's ok because of router still running");
+            });
+
+//          var histogram = client.getMetricsClient().getReconnectDurationHistogram().get();
+//          logger.info("sample count {}", histogram.getSampleCount());
+//          logger.info("sample sum {}", histogram.getSampleSum());
+//          histogram.getBuckets().forEach(b -> {
+//              logger.info(b.toString());
+//          });
+
+            var optional = client.getMetricsClient().getReconnectDurationHistogram();
+            if (optional.isPresent()) {
+                var histogram = optional.get();
+                double average = histogram.getSampleSum() / histogram.getSampleCount();
+                averages.add(average);
+            }
+
+        }
+
+        double sum = 0;
+        for (var a : averages) {
+            sum += a;
+        }
+        double average = sum / averages.size();
+        return Duration.ofSeconds((long) average);
```

Comment:
I would recommend that we use a percentile such as 99%, as that is easiest to give guarantees about. Averages can lie and hide real issues. You should be able to get that from the histogram bucket for that percentile.

Reply:
I have just addressed this comment, wdyt?